test/integration/test_analysis_graph.c

```diff
@@ -0,0 +1,92 @@
+// SPDX-FileCopyrightText: 2022 billow <billow.fun@gmail.com>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_core.h>
+#include <rz_analysis.h>
+#include <rz_project.h>
+
+#include "../unit/minunit.h"
+
+static inline RzGraph *graph_by_function_name(RzCore *core, const char *name) {
+    RzAnalysisFunction *f = rz_analysis_get_function_byname(core->analysis, name);
+    char *msg = rz_str_newf("find function %s", name);
+    mu_assert_notnull(f, msg);
+    RZ_FREE(msg);
+
+    RzGraph *g = rz_core_graph(core, RZ_CORE_GRAPH_TYPE_DATAREF, f->addr);
+    msg = rz_str_newf("create graph %s", name);
+    mu_assert_notnull(g, msg);
+    free(msg);
+    return g;
+}
+
+bool test_analysis_graph() {
+    // 1. Open the file
+    RzCore *core = rz_core_new();
+    mu_assert_notnull(core, "new RzCore instance");
+    ut64 loadaddr = 0;
+    const char *fpath = "bins/elf/ls";
+    RzCoreFile *file = rz_core_file_open(core, fpath, RZ_PERM_R, loadaddr);
+    mu_assert_notnull(file, "open file");
+    rz_core_bin_load(core, fpath, loadaddr);
+
+    // 2. Analyse the file
+    rz_core_analysis_all(core);
+    rz_core_analysis_everything(core, false, "esil");
+
+    // 3. dataref graph
+    RzGraph *g = graph_by_function_name(core, "entry0");
```

Comment:
Can you use some more complex functions than with just one node? Also, maybe multiple graph types at once: CFG, callgraph, refs.